### PR TITLE
Factory loading flow refactoring

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR splits the whole factory loading functionality into separate components that represent individual steps of the loading flow: [StepInitialize](https://github.com/eclipse-che/che-dashboard/tree/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Initialize/index.tsx), [StepFetchResources](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/FetchResources/index.tsx), [StepFetchDevfile](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/FetchDevfile/index.tsx), [StepCreateWorkspace](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/CreateWorkspace/index.tsx), [StepApplyResources](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/ApplyResources/index.tsx), and [StepApplyDevfile](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Factory/Steps/ApplyDevfile/index.tsx). 

The other goal of this refactoring was to align factory and workspaces loading flows. It allowed delegating the two last steps of the factory flow ([StepStartWorkspace](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Workspace/Steps/StartWorkspace/index.tsx) and [StepOpenWorkspace](https://github.com/eclipse-che/che-dashboard/blob/refactor-factory-loading/packages/dashboard-frontend/src/containers/Loader/Workspace/Steps/OpenWorkspace/index.tsx)) to the workspace loader.

### What issues does this PR fix or reference?

resolves https://github.com/eclipse/che/issues/21449
fixes https://github.com/eclipse/che/issues/21308

### Is it tested? How?

The PR can be tested here: https://eclipse-che.apps.cluster-h8r9f.h8r9f.sandbox1016.opentlc.com/

### CODE REVIEW

File: packages/dashboard-frontend/src/services/registry/resources.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      e,
+    )}`;
+    console.error(errorMessage);
+    throw new Error(errorMessage);
+  }
+}
+
+export function loadResourcesContent(data: string): PreBuiltResources {
+  if (!data) {
+    throw new Error(`Pre-built resources content is empty.`);
```
Comment (olexii4) [inline packages/dashboard-frontend/src/services/registry/resources.ts:34]: `Pre-built resources content is empty.` -> 'Pre-built resources content is empty.'

**Comment by olexii4** [comment]
I have a stable bug with this solution - just try to stop the target workspace from a separate tab:
Знімок екрана 2022-07-25 о 17 20 35

File: packages/dashboard-frontend/src/services/registry/resources.ts (hunk 1 above)
Comment (olexii4) [inline packages/dashboard-frontend/src/services/registry/resources.ts:34]:
```suggestion
    throw new Error('Pre-built resources content is empty.');
```

**Review by olexii4 — APPROVED** [review]
I tested this PR on both platforms(minikube and OpenShift). It works as expected. 
Great job!

**Review by ibuziuk — APPROVED** [review]
Great Job! great job
verified locally against dogfooding
It feel like the startup is much faster now

**Pull request merged**